azurerm/resource_arm_netapp_volume_test.go

```diff
@@ -0,0 +1,322 @@
+		accountName := rs.Primary.Attributes["account_name"]
+		poolName := rs.Primary.Attributes["pool_name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		client := testAccProvider.Meta().(*ArmClient).Netapp.VolumeClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		if resp, err := client.Get(ctx, resourceGroup, accountName, poolName, name); err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: NetApp Volume %q (Resource Group %q) does not exist", name, resourceGroup)
+			}
+			return fmt.Errorf("Bad: Get on netapp.VolumeClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMNetAppVolumeDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*ArmClient).Netapp.VolumeClient
+	ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_netapp_volume" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		accountName := rs.Primary.Attributes["account_name"]
+		poolName := rs.Primary.Attributes["pool_name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		if resp, err := client.Get(ctx, resourceGroup, accountName, poolName, name); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Get on netapp.VolumeClient: %+v", err)
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMNetAppVolume_basic(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-netapp-%d"
+  location = "%s"
+}
+
+resource "azurerm_virtual_network" "test" {
+  name                = "acctest-VirtualNetwork-%d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  address_space       = ["10.0.0.0/16"]
+}
+
+resource "azurerm_subnet" "test" {
+  name                 = "acctest-Subnet-%d"
+  resource_group_name  = "${azurerm_resource_group.test.name}"
+  virtual_network_name = "${azurerm_virtual_network.test.name}"
+  address_prefix       = "10.0.2.0/24"
+  delegation {
+    name = "testdelegation"
+
+    service_delegation {
+      name    = "Microsoft.Netapp/volumes"
+      actions = ["Microsoft.Network/networkinterfaces/*", "Microsoft.Network/virtualNetworks/subnets/join/action"]
+    }
+  }
+}
+
+resource "azurerm_netapp_account" "test" {
+  name                = "acctest-NetAppAccount-%d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+}
+
+resource "azurerm_netapp_pool" "test" {
+  name                = "acctest-NetAppPool-%d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  account_name        = "${azurerm_netapp_account.test.name}"
+  service_level       = "Premium"
+  size_in_4_tb        = "1"
+}
```

Comment:
rather than defining these the same for all of the tests we're using [template functions](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_windows_virtual_machine_scale_set_test.go#L67) to remove the commonality - can we update this?

Reply:
fixed